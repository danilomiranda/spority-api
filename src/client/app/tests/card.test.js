/* global it expect */
import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'

import Card from '../components/card'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Card
          key='3Rvw51S5aB7G87bLz5hw1t'
          {...{
            id: '3Rvw51S5aB7G87bLz5hw1t',
            name: 'SÃ©culo Sinistro',
            images: [
              {
                height: 640,
                url:
                  'https://i.scdn.co/image/8f8fac6ace8d5b206e23101c5ecfe70015fba27e',
                width: 640
              },
              {
                height: 300,
                url:
                  'https://i.scdn.co/image/ec9b711714a557bb60e4785cdf718f446691a798',
                width: 300
              },
              {
                height: 64,
                url:
                  'https://i.scdn.co/image/27ae3aefe740b161a6baeb5450126be585b3cc77',
                width: 64
              }
            ]
          }}
        />
      </MemoryRouter>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
