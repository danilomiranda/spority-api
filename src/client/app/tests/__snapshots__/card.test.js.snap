// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="col-2"
>
  <div
    className="media-object"
  >
    <div
      className="media-object-hoverable"
    >
      <div
        className="image-wrapper"
      >
        <img
          className="cover-art-image"
          src="https://i.scdn.co/image/ec9b711714a557bb60e4785cdf718f446691a798"
        />
      </div>
      <div
        className="mo-info"
      >
        <div
          className="react-contextmenu-wrapper"
        >
          <a
            className="mo-info-name"
            href="/album/3Rvw51S5aB7G87bLz5hw1t"
            onClick={[Function]}
            title="Brazilian Pop"
          >
            SÃ©culo Sinistro
          </a>
        </div>
      </div>
    </div>
    <div
      className="mo-meta ellipsis-one-line"
    >
      <div
        className="react-contextmenu-wrapper"
      >
        <a
          href="/artist/2K13AVg3bFpHSxDM1vJ0qA"
        />
      </div>
    </div>
  </div>
  <style
    jsx="true"
  >
    
      .media-object {
        position: relative;
        padding-bottom: 2.5em;
        max-width: 450px;
      }
      .cover-art-image {
        transition-property: opacity;
        transition-duration: .3s;
        transition-timing-function: cubic-bezier(.3,0,.4,1);
        position: absolute;
        top: 0;
        width: 100%;
        height: 100%;
        background-size: cover!important;
        background-position: 50%;
        opacity: 1;
      }
      .image-wrapper {
        width: 188px;
        height: 188px;
        position: relative;
        margin: 0 8px 8px 0;
        float: left;
      }
      .mo-info {
        margin: 12px 0 4px;
        width: 100%;
        overflow: hidden;
        text-overflow: ellipsis;
        text-align: center;
        display: block;
        display: -webkit-box;
        -webkit-line-clamp: 2;
        -webkit-box-orient: vertical;
      }
      .react-contextmenu-wrapper {
        display: inline;
      }
      .mo-info-name {
        font-size: 14px;
        line-height: 20px;
        letter-spacing: .015em;
        font-weight: 400;
        color: #fff;
        text-transform: none;
        max-height: 560px;
      }
      .media-object a {
        border-bottom-width: 0;
      }
    
  </style>
  <style
    jsx="true"
  >
    
      .col-2 {
        position: relative;
        width: 100%;
        min-height: 1px;
        padding-right: 15px;
        padding-left: 15px;
        max-width: none;
        -ms-flex: 0 0 16.666667%;
        flex: 0 0 16.666667%;
        max-width: 16.666667%;
        min-width: 190px;
      }
      .col-12 {
        position: relative;
        width: 100%;
        min-height: 1px;
        padding-right: 15px;
        padding-left: 15px;
        -ms-flex: 0 0 100%;
        flex: 0 0 100%;
        max-width: 100%;
      }
    
  </style>
</div>
`;
